src/xrpld/app/tx/detail/apply.cpp

```diff
@@ -113,12 +146,88 @@ apply(
     ApplyFlags flags,
     beast::Journal j)
 {
-    STAmountSO stAmountSO{view.rules().enabled(fixSTAmountCanonicalize)};
-    NumberSO stNumberSO{view.rules().enabled(fixUniversalNumber)};
+    return apply(app, view, [&]() mutable {
+        return preflight(app, view.rules(), tx, flags, j);
+    });
+}
 
-    auto pfresult = preflight(app, view.rules(), tx, flags, j);
-    auto pcresult = preclaim(pfresult, app, view);
-    return doApply(pcresult, app, view);
+std::pair<TER, bool>
+apply(
+    Application& app,
+    OpenView& view,
+    uint256 const& batchId,
+    STTx const& tx,
+    ApplyFlags flags,
+    beast::Journal j)
+{
+    return apply(app, view, [&]() mutable {
+        return preflight(app, view.rules(), batchId, tx, flags, j);
+    });
+}
+
+static bool
+applyBatchTransactions(
+    Application& app,
+    OpenView& batchView,
+    STTx const& txn,
+    beast::Journal j)
+{
+    XRPL_ASSERT(
+        txn.getTxnType() == ttBATCH &&
+            txn.getFieldArray(sfRawTransactions).size() != 0,
+        "Batch transaction missing sfRawTransactions");
+
+    auto const batchId = txn.getTransactionID();
+    auto const mode = txn.getFlags();
+
+    auto applyOneTransaction = [&app, &j, &batchId, &batchView](STTx&& tx) {
+        OpenView perTxBatchView(batch_view, batchView);
+
+        auto const ret = apply(app, perTxBatchView, batchId, tx, tapBATCH, j);
+        XRPL_ASSERT(
+            ret.second == (isTesSuccess(ret.first) || isTecClaim(ret.first)),
```

Comment:
Why does this allow `tecCLAIM` as well? From the description, it looks like we shouldn't be executing the inner transactions in that case.

Reply:
bad comment, should be "Inner transaction should not be applied"